File: src/main/java/dev/elysium/eraces/commands/MyraceCommand.java
package dev.elysium.eraces.commands;

import com.mojang.brigadier.Command;
import com.mojang.brigadier.tree.LiteralCommandNode;
import dev.elysium.eraces.ERaces;
import dev.elysium.eraces.utils.ChatUtil;
import io.papermc.paper.command.brigadier.CommandSourceStack;
import io.papermc.paper.command.brigadier.Commands;
import lombok.Getter;
import org.bukkit.entity.Player;

import java.util.Map;

public class MyraceCommand {
    @Getter
    private final LiteralCommandNode<CommandSourceStack> cmd;

    public MyraceCommand() {
        this.cmd = Commands.literal("myrace")
                .requires(source -> source.getSender() instanceof Player)
                .executes(ctx -> {
                    myraceCommandLogic((Player) ctx.getSource().getSender());
                    return Command.SINGLE_SUCCESS;
                }).build();
    }

    private static void myraceCommandLogic(Player player) {
        var context = ERaces.getInstance().getContext();

        String playerRace = context.getPlayerDataManager().getPlayerRaceId(player);

        if (playerRace == null || playerRace.isEmpty()) {
            ChatUtil.message(player, context.getMessageManager().getData().getRaceNotSelected());
            return;
        }

        String message = context.getMessageManager().getData().getGetPlayerRaceSuccessMe();
        ChatUtil.message(player, message, Map.of("{race}", playerRace));
    }
}

File: src/main/java/dev/elysium/eraces/commands/MyRaceCommand.kt
package dev.elysium.eraces.commands

import com.mojang.brigadier.Command
import com.mojang.brigadier.context.CommandContext
import com.mojang.brigadier.tree.LiteralCommandNode
import dev.elysium.eraces.ERaces.Companion.getInstance
import dev.elysium.eraces.utils.msg
import io.papermc.paper.command.brigadier.CommandSourceStack
import io.papermc.paper.command.brigadier.Commands
import org.bukkit.entity.Player

object MyRaceCommand {
    val cmd: LiteralCommandNode<CommandSourceStack> =
        Commands.literal("myrace").requires { source: CommandSourceStack -> source.sender is Player }
            .executes { ctx: CommandContext<CommandSourceStack> ->
                myRaceCommandLogic(ctx.getSource().sender as Player)
                Command.SINGLE_SUCCESS
            }.build()


    private fun myRaceCommandLogic(player: Player) {
        val messages = getInstance().context.messageManager.data
        val playerRace = getInstance().context.playerDataManager.getPlayerRaceId(player)

        if (playerRace.isNullOrEmpty()) {
            player.msg(messages.raceNotSelected)
            return
        }

        val message = messages.getPlayerRaceSuccessMe
        player.msg(message, Pair("{race}", playerRace))
    }
}